src/nuclide_header.F90

```diff
@@ -265,11 +267,31 @@ subroutine nuclide_from_hdf5(this, group_id, temperature, method, tolerance)
                &for " // trim(this % name) // " at or near " // &
                trim(to_str(nint(temp_desired))) // " K.")
         end if
-      end do TEMP_LOOP
+      end do
 
     case (TEMPERATURE_INTERPOLATION)
-      ! TODO: Get bounding temperatures
-      call fatal_error("Temperature interpolation not yet implemented")
+      ! If temperature interpolation or multipole is selected, get a list of
+      ! bounding temperatures for each actual temperature present in the model
+      TEMP_LOOP: do i = 1, temperature % size()
+        temp_desired = temperature % data(i)
+
+        do j = 1, size(temps_available) - 1
```

Comment:
So what happens if the user selects the global temperature interpolation option, but provides a dataset with only one temperature (like our NNDC set)?  If the desired temperature is exactly the library temperature it should run no problem and if not, then this interpolation should fail.  Is this the actual behavior?